var/spack/repos/builtin/packages/py-radical-pilot/package.py

```diff
@@ -13,44 +13,102 @@ class PyRadicalPilot(PythonPackage):
+        "1.18.1",
+        sha256="fd6a0ffaa727b6b9bab35d8f2dc300bf4d9c4ff3541136d83560aa7b853d6100",
+        deprecated=True,
+    )
+    version(
+        "1.17.0",
+        sha256="0bfbb321a623a684e6694241aa3b7804208846515d23afa3b930553274f4a69f",
+        deprecated=True,
+    )
+    version(
+        "1.16.0",
+        sha256="057941a206ee96b62b97a63a507c1136b7fe821ae9f9e5eebe7949a3f53941f9",
+        deprecated=True,
+    )
+    version(
+        "1.15.1",
+        sha256="35c3b179a0bc85f52d2165e98e19acf2bf79037dd14f4d9ff3fc55ae0122d17e",
+        deprecated=True,
+    )
+    version(
+        "1.14.0",
+        sha256="462471065de25f6d6e8baee705790828444c2eebb2073f5faf67a8da800d15a9",
+        deprecated=True,
+    )
+    version(
+        "1.13.0",
+        sha256="5bd9eef1884ccca09c242ab6d1361588a442d9cd980613c66604ba140786bde5",
+        deprecated=True,
+    )
+    version(
+        "1.12.0",
+        sha256="a266355d30d838f20b6cac190ce589ca919acd41883ad06aec62386239475133",
+        deprecated=True,
+    )
+    version(
+        "1.11.2",
+        sha256="9d239f747589b8ae5d6faaea90ea5304b6f230a1edfd8d4efb440bc3799c8a9d",
+        deprecated=True,
+    )
+    version(
+        "1.10.2",
+        sha256="56e9d8b1ce7ed05eff471d7df660e4940f485027e5f353aa36fd17425846a499",
+        deprecated=True,
+    )
+    version(
+        "1.10.1",
+        sha256="003f4c519b991bded31693026b69dd51547a5a69a5f94355dc8beff766524b3c",
+        deprecated=True,
+    )
+    version(
+        "1.9.2",
+        sha256="7c872ac9103a2aed0c5cd46057048a182f672191e194e0fd42794b0012e6e947",
+        deprecated=True,
+    )
+    version(
+        "1.8.0",
+        sha256="a4c3bca163db61206e15a2d820d9a64e888da5c72672448ae975c26768130b9d",
+        deprecated=True,
+    )
+    version(
+        "1.6.8",
+        sha256="fa8fd3f348a68b54ee8338d5c5cf1a3d99c10c0b6da804424a839239ee0d313d",
+        deprecated=True,
+    )
+    version(
+        "1.6.7",
+        sha256="6ca0a3bd3cda65034fa756f37fa05681d5a43441c1605408a58364f89c627970",
+        deprecated=True,
+    )
+
+    depends_on("py-radical-gtod", type=("build", "run"))
     depends_on("py-radical-saga", type=("build", "run"))
-    depends_on("py-radical-gtod", type=("build", "run"), when="@1.14:")
 
-    depends_on("py-radical-utils@1.12:", type=("build", "run"), when="@1.12:")
-    depends_on("py-radical-saga@1.12:", type=("build", "run"), when="@1.12:")
+    depends_on("py-radical-utils@1.46:", type=("build", "run"), when="@1.46:")
 
-    depends_on("py-radical-utils@1.8.4:1.11", type=("build", "run"), when="@1.11")
-    depends_on("py-radical-saga@1.8:1.11", type=("build", "run"), when="@1.11")
+    depends_on("py-radical-utils@1.39", type=("build", "run"), when="@1.39")
 
-    depends_on("py-radical-utils@:1.8.3", type=("build", "run"), when="@:1.10")
-    depends_on("py-radical-saga@:1.7", type=("build", "run"), when="@:1.10")
+    depends_on("py-pymongo@:3", type=("build", "run"), when="@1.39")
 
```

Comment:
Missing dependency on `py-radical-saga@1.37:`. If you sort your dependencies in the same order as requirements.txt, it makes these PRs much easier to review.

Reply:
SAGA dependency is set earlier (`depends_on("py-radical-saga", type=("build", "run"))`), since we currently support only two versions 1.39 and 1.4x, I didn't set that version requirement, should I bring it in?

Reply:
Yes, it's always good to be explicit about version requirements in case someone adds an older version in the future.